### DISCUSSION TITLE
graphqlGenerateTestClient requires compileKotlin

### DISCUSSION THREAD

**Discussion by hrkfdn** [ExpediaGroup/graphql-kotlin#1123] [Q&A] [answered]
Hi there,

I am about to migrate to the latest release (4.0.0), but I am having problems with client generation in the test scope.

We are using the client within the test scope, so we are using the `graphqlGenerateTestClient` task. Previously this worked fine, but with 4.0.0 it seems that `compileKotlin` is also trying to compile the test clients. This fails, because `graphql-kotlin-client` is only included in the test scope:
```gradle
testImplementation "com.expediagroup:graphql-kotlin-client:$expediaGraphQL"
```

Thus, we are getting some compile errors during `compileKotlin`:
```
e: /path/TestQuery.kt: (13, 5): Unresolved reference: GraphQLClientRequest
```

Did the project scope for the generated test clients somehow change in 4.x?

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
We rewrote the Gradle plugin to use the worker API that helped isolate the classpaths. This sounds like a bug (its probably pulling both regular and test client tasks here -> https://github.com/ExpediaGroup/graphql-kotlin/blob/master/plugins/graphql-kotlin-gradle-plugin/src/main/kotlin/com/expediagroup/graphql/plugin/gradle/GraphQLGradlePlugin.kt#L140). While I'm looking into this, by any chance can you share a link to a repo that has this problem?

**Reply by dariuszkuc** [reply]
Fixed in https://github.com/ExpediaGroup/graphql-kotlin/pull/1124

**Reply by hrkfdn** [reply]
Sorry, would have loved to provide an example but you were way faster :) Thanks for the fix!
